ros2/ubuntu/Dockerfile

```
@@ -1,44 +1,85 @@
+RUN echo "yaml file:///etc/ros/rosdep/prereqs.yaml" > /etc/ros/rosdep/sources.list.d/10-custom.list
+RUN rosdep update
+
+# setup new entrypoint and default command
+COPY orocos_entrypoint.sh /orocos_entrypoint.sh
+ENTRYPOINT [ "/orocos_entrypoint.sh" ]
+CMD [ "bash" ]
+
+## *************************************************************************************************
+## Stage orocos_toolchain: clone, build and install Orocos Toolchain using CMake (with CORBA)
+## *************************************************************************************************
+FROM base AS orocos_toolchain
+
+COPY orocos_toolchain.repos /build/
+RUN vcs import --recursive /build < /build/orocos_toolchain.repos \
+    && cd /build/orocos_toolchain \
+    && . /opt/ros/${ROS_DISTRO}/local_setup.sh \
+    #
+    # Install dependencies with rosdep
+    && apt-get update \
+    && apt-get install -y ruby \
+    && rosdep install -y --from-path . --ignore-src \
+    && rm -rf /var/lib/apt/lists/* \
+    #
+    # Build the workspace
+    && colcon build \
         --event-handlers console_direct+ \
-        --install-base /opt/ros/${ROS_DISTRO} \
+        --install-base /opt/orocos/${ROS_DISTRO} \
         --merge-install \
         --cmake-args \
+            \ -DBUILD_TESTING=OFF \
             \ -DCMAKE_BUILD_TYPE=Release \
-            \ -DCMAKE_FIND_ROOT_PATH=/stage \
-            \ -DCMAKE_PREFIX_PATH=/opt/ros/${ROS_DISTRO} \
             \ -DENABLE_CORBA=ON \
             \ -DCORBA_IMPLEMENTATION=OMNIORB \
-            \ -DOROCOS_INSTALL_INTO_PREFIX_ROOT=ON
+            \ -DOROCOS_INSTALL_INTO_PREFIX_ROOT=ON \
+    #
+    # Cleanup
+    && rm -rf /build
 
-## Stage deploy: copy the install-space to /opt/ros/${ROS_DISTRO} as a
-## new image layer based on deps
-FROM deps AS deploy
-COPY --from=build /stage/opt/ros/${ROS_DISTRO} /opt/ros/${ROS_DISTRO}
+## *************************************************************************************************
+## Stage overlay: clone, build and install overlay workspaces to /opt/ros/${ROS_DISTRO}
+## *************************************************************************************************
+FROM orocos_toolchain AS overlay
 
-# default config file for colcon
-COPY .colcon/defaults.yaml /root/.colcon/defaults.yaml
+COPY overlay.repos /build/
+RUN mkdir -p /build/src \
+    && cd /build \
+    && vcs import --recursive src < overlay.repos \
+    && . /opt/ros/${ROS_DISTRO}/local_setup.sh \
+    && . /opt/orocos/${ROS_DISTRO}/local_setup.sh \
+    #
+    # Install dependencies with rosdep
+    && apt-get update \
+    && ROS_PACKAGE_PATH=/opt/orocos/${ROS_DISTRO}/share:/opt/ros/${ROS_DISTRO}/share \
+           rosdep install -y --from-path . --ignore-src \
+    && rm -rf /var/lib/apt/lists/* \
+    #
+    # Build the workspace
+    && colcon build \
+        --event-handlers console_direct+ \
+        --install-base /opt/orocos/${ROS_DISTRO} \
+        --merge-install \
+        --parallel-workers 2 \
```

My machine with 16 GB of memory still fails in these conditions:
```suggestion
        --parallel-workers 1 \
```
But CI systems may have more resources.

---

I think Docker Hub will not work even with `--parallel-workers 1` because the workers are quite limited in resources according to [this post](https://success.docker.com/article/what-are-the-current-resource-limits-placed-on-automated-builds):

> The current limits placed on Automated Builds are:
> 
> 2 hours
> 2 GB RAM
> 1 CPU
> 30 GB Disk Space

So probably we have to switch to Travis or manual builds. Travis [today](https://docs.travis-ci.com/user/reference/overview/) offers 7.5 GB of memory and 2 cores for Ubuntu builders, which should be enough.

---

Removed in https://github.com/orocos/orocos-docker-images/pull/3/commits/dd58443825930692ae6f3b714f7de5d50c0fed0f.